### PR TITLE
Fix cert with rsa instead of rsaEncryption as public key algorithm

### PR DESCRIPTION
**Pull request by bernd-edlinger**
##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: crypto/rsa/rsa_ameth.c [hunk 1]
```diff
@@ -34,7 +34,7 @@ static int rsa_param_encode(const EVP_PKEY *pkey,
 
     *pstr = NULL;
     /* If RSA it's just NULL type */
-    if (pkey->ameth->pkey_id == EVP_PKEY_RSA) {
+    if (pkey->ameth->pkey_id != EVP_PKEY_RSA_PSS) {
```
Comment (kroeckx) [inline crypto/rsa/rsa_ameth.c]: I wonder if it makes sense to use `pkey_id == EVP_PKEY_RSA || pkey_id == EVP_PKEY_RSA2`. I assume that at some point we'll add also add OAEP, but I guess we can worry about that when we add it.

Comment (bernd-edlinger) [reply crypto/rsa/rsa_ameth.c]: Yes, as a side-effect this should allow a workaround for rsaOaep (without actually
implementing the correct Oaep parameter structure, see #7907)

**Comment by bernd-edlinger** [comment]
Sorry, I forgot to add the 1.1.1 label,
is your approval good for 1.1.1 as well?

**Comment by kroeckx** [comment]
Yes. I assume 1.1.0 doesn't have this problem.

**Comment by bernd-edlinger** [comment]
Exactly, it is a 1.1.1 regression.

**Comment by bernd-edlinger** [comment]
Merged to master as 1f483a6, and 1.1.1 as c25ae0f.
Thanks!

**Pull request closed without merging**